backend/management/commands/config.py

```diff
@@ -144,23 +159,14 @@ def set_backend_config(self, backend_slug, config_yaml):
             if "query" not in example:
                 raise Exception(f"No `query` key found in example #{index}")
 
-        # Check that the `id` and `slug` of the backend match `backend_id` and
-        # `backend_slug`, respectively.
-        backend_config_id = backend_config.get("id", None)
+        # Check that the `slug` specified in the YAML file matches the slug
+        # with which this command was called.
         backend_config_slug = backend_config.get("slug", None)
-        try:
-            backend_config_id = int(backend_config_id)
-        except Exception:
-            raise Exception(f"Id `{backend_config_id}` is not an integer")
-        if backend_config_id != backend_id:
-            raise Exception(
-                f"Id {backend_config_id} from YAML does not match backend id "
-                f"{backend_id} obtained via slug `{backend_slug}`"
-            )
         if backend_config_slug != backend_slug:
             raise Exception(
-                f"Slug {backend_config_slug} from YAML does not match backend "
-                f"slug `{backend_slug}`"
+                f"This command was called for backend slug `{backend_slug}`, "
+                f"but the value for `slug` in the YAML file is "
+                f"`{backend_config_slug}`; the two slugs must match"
             )
```

Comment:
This is an assertion rather than catching an exception that expect. The message could reflect that this is an internal error.

Reply:
@Qup42 You said so during our meeting already but I don't understand it. You can call this command with two arguments: the backend slug and a YAML file. So it can happen that the YAML file mentions a different slug. Of course, that is not how the command should be called, but it can happen.

Reply:
Thanks for clearing that up. I was thinking of `get_backend_config` where this should indeed not happen. But here it is of course a normal error.